internal/handler/impl/domain_handler.go

```diff
@@ -45,18 +45,19 @@ func (a *application) ListDomains(
 		tx     *gorm.DB
 		xrhid  *identity.XRHID
 	)
-	log := app_context.LogFromCtx(ctx.Request().Context())
+	handlerName := "ListDomains"
+	logger := app_context.LogFromCtx(ctx.Request().Context())
 	if xrhid, err = getXRHID(ctx); err != nil {
-		log.Error(err.Error())
+		logger.Error(errXRHIDIsNil, slog.String("handler", handlerName))
 		return err
 	}
 	// TODO A call to an internal validator could be here to check public.ListTodosParams
 	if orgID, offset, limit, err = a.domain.interactor.List(xrhid, &params); err != nil {
-		log.Error(err.Error())
+		logger.Error("input adapter", slog.String("handler", handlerName))
 		return err
 	}
 	if tx = a.db.Begin(); tx.Error != nil {
-		log.Error(tx.Error.Error())
+		logger.Error("starting database transaction", slog.String("handler", handlerName))
```

Comment:
As we discussed on mtg. It would be good if all the error messages were written in a way that describe that something failed. E.g. instead of "starting database transaction" it would say: "failed to begin database transaction".  I.e. reporting the outcome.

Applies for a lot of the other messages in this PR.